package/src/components/ChannelList/ChannelList.tsx

```diff
@@ -290,8 +281,68 @@ export const ChannelList = <
   } = props;
 
   const [forceUpdate, setForceUpdate] = useState(0);
-  const { enableOfflineSupport } = useChatContext<StreamChatGenerics>();
+  const { client, enableOfflineSupport } = useChatContext<StreamChatGenerics>();
+  const channelManager = useMemo(() => client.createChannelManager({}), [client]);
+
+  /**
+   * This hook sets the event handler overrides in the channelManager internally
+   * whenever they change. We do this to avoid recreating the channelManager instance
+   * every time these change, as we want to keep it as static as possible.
+   * This protects us from something like defining the overrides as inline functions
+   * causing the manager instance to be recreated over and over again.
+   */
+  useEffect(() => {
+    channelManager.setEventHandlerOverrides({
+      channelDeletedHandler: onChannelDeleted,
+      channelHiddenHandler: onChannelHidden,
+      channelTruncatedHandler: onChannelTruncated,
+      channelVisibleHandler: onChannelVisible,
+      memberUpdatedHandler: onChannelMemberUpdated
+        ? (setChannels, event) =>
+            onChannelMemberUpdated(lockChannelOrder, setChannels, event, { filters, sort })
+        : undefined,
+      newMessageHandler: onNewMessage
+        ? (setChannels, event) =>
+            onNewMessage(lockChannelOrder, setChannels, event, { filters, sort })
+        : undefined,
+      notificationAddedToChannelHandler: onAddedToChannel
+        ? (setChannels, event) => onAddedToChannel(setChannels, event, { filters, sort })
+        : undefined,
+      notificationNewMessageHandler: onNewMessageNotification
+        ? (setChannels, event) => onNewMessageNotification(setChannels, event, { filters, sort })
+        : undefined,
+      notificationRemovedFromChannelHandler: onRemovedFromChannel,
+    });
+  }, [
+    channelManager,
+    filters,
+    lockChannelOrder,
+    onAddedToChannel,
+    onChannelDeleted,
+    onChannelHidden,
+    onChannelMemberUpdated,
+    onChannelTruncated,
+    onChannelVisible,
+    onNewMessage,
+    onNewMessageNotification,
+    onRemovedFromChannel,
+    sort,
+  ]);
+
+  useEffect(() => {
+    channelManager.setOptions({ abortInFlightQuery: true, lockChannelOrder });
```

Comment:
What does `abortInFlightQuery` do?

Reply:
It essentially bypasses the `loading` state if another `channelManager.queryChannels` is triggered. We only do this on the RN SDK and not on the others, hence why I added it here. 

I'm generally against this type of stuff, but this seems to be some core functionality as seen in [this test](https://github.com/GetStream/stream-chat-react-native/blob/f79bc2b27bb1649de324324ebe65b0a2574dc282/package/src/components/ChannelList/__tests__/ChannelList.test.js#L147). I added it just to be on the safe side and would really like to gradually discontinue it at some point (maybe add it as an optional `prop` first, then deprecate the behaviour and then maybe finally remove it). The reason why is because it can easily be a foot gun, it's really only here for backwards compatibility